File: backend/code-challenges/java/src/test/java/LoanMatcherTest.java
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class LoanMatcherTest {

    @Test
    void returnsPersonalLoanWhenCustomerIncomeIsUnder3000() {
        Customer customer = new Customer(3000);
        LoanMatcher loanMatcher = new LoanMatcher(customer);
        List<Loan> availableLoans = loanMatcher.loans();

        assertThat(availableLoans.size()).isEqualTo(1);

        assertThat(availableLoans.stream().findFirst().isPresent()).isTrue();
        assertThat(availableLoans.get(0).type()).isEqualTo("PERSONAL_LOAN");
    }
}

File: backend/code-challenges/kotlin/src/test/kotlin/LoanMatcherTest.kt
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class LoanMatcherTest {

    @Test
    internal fun `returns personal loan when customer income is under 3000`() {
        val customer = Customer(3000)
        val loanMatcher = LoanMatcher(customer)
        val availableLoans: List<Loan> = loanMatcher.loans()

        assertThat(availableLoans.size).isEqualTo(1)

        assertThat(availableLoans.stream().findFirst().isPresent).isTrue
        assertThat(availableLoans[0].type()).isEqualTo("EMPRESTIMO_PESSOAL")
    }
}